spec/libraries/materials_spec.rb

```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+RSpec.describe 'Extension: Materials' do
+  class self::Assessment < ActiveRecord::Base
+    def self.columns
+      []
+    end
+
+    has_one_folder
+  end
+
+  describe self::Assessment do
+    it { is_expected.to have_one(:folder) }
+    it { is_expected.to have_many(:materials) }
+
+    let(:files) { [OpenStruct.new(original_filename: 'file.txt')] }
+    let(:assessment) { self.class::Assessment.new }
+    describe 'files_attributes=' do
+      before do
+        assessment.build_folder(name: 'folder')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.